feat: add wasm watch and some docs on set up (#3572)
marimo-team/marimo: docs/guides/editor_features/watching.md

!!! note
    Support for watching data files and automatically refreshing cells that depend on them is coming soon. Follow along at <https://github.com/marimo-team/marimo/issues/3258>

## Developing in WASM mode

> You can develop a notebook locally and preview it as a WASM application. This lets you take advantage of local development tools while seeing the notebook as it will appear when deployed to production.

Reviewer: ```suggestion
Follow these steps to develop a notebook using your own editor, while previewing it as a WebAssembly notebook in the browser. This lets you take advantage of local development tools while seeing the notebook as it will appear when deployed to production as a WebAssembly notebook.
```

Also, link "WebAssembly notebook" to the user guide on WebAssembly notebooks?

```bash

> # in one terminal, edit your notebook here / editor of choice

Reviewer: ```suggestion
# in one terminal, start a watched edit session
```
